components/formats-gpl/src/loci/formats/in/LIFReader.java

```diff
@@ -1947,7 +2020,7 @@ else if (unit.equals("m")) {
             if (ms.sizeZ == 1) {
               ms.sizeZ = len;
               bytesPerAxis.put(nBytes, "Z");
-              physicalSizeZ = (physicalLen * len) / (len - 1);
+              physicalSizeZ = physicalLen;
```

Comment:
I agree that what you describe is the most intuitive way to interpret sizes. The decision to not store it like that in lif-files was probably driven by the desire to make no distinction between x, y and z dimensions.
And since setting z-size to a nonzero value for "2D" images would probably confuse users quite a lot (even if that would arguably be the most honest representation), treating x and y similar to z was the route that was chosen.